WordPress/Classes/Services/NotificationFilteringService.swift

```diff
@@ -24,6 +24,10 @@ final class NotificationFilteringService {
 
         set {
             userDefaults?.set(newValue, forKey: WPNotificationsEnabledKey)
+
+            if isWordPress {
```

Comment:
We should only cancel when we disable

```suggestion
            if isWordPress && !newValue {
```

Reply:
Thanks for finding this + agree.  I have pushed the code change for this